testutil/signer/private.go

```diff
@@ -0,0 +1,43 @@
+package signer
+
+import (
+	"os"
+
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	cosmosed "github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+
+	"github.com/babylonlabs-io/babylon/app/signer"
+	"github.com/babylonlabs-io/babylon/privval"
+	checkpointingtypes "github.com/babylonlabs-io/babylon/x/checkpointing/types"
+)
+
+// SetupTestPrivSigner sets up a PrivSigner for testing
+func SetupTestPrivSigner() (*signer.PrivSigner, error) {
+	// Create a temporary node directory
+	nodeDir, err := os.MkdirTemp("", "tmp-signer")
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		_ = os.RemoveAll(nodeDir)
+	}()
```

Comment:
```suggestion
	nodeDir := t.TempDir()
```

Reply:
t.TempDir() internally calls t.CleanUp, so a bit cleaner

Reply:
problem is, I don't have the `*testing.T` there